.github/changelog.py

```diff
@@ -0,0 +1,35 @@
+import sys
+
+"""
+This script assumes changelogs use the following format:
+    ## <major>.x (eg. "## 15.x") to indicate a major version series
+    ### <major>.<minor>-<suffix> <date etc> (eg. "## 15.0 (2025-04-01)", "### 15.1-beta1 (2024-12-25)") to indicate an individual version
+"""
+
+def main():
+    current_version = sys.argv[1]
+    stable_version = current_version.split("-")[0]
+    major_version = current_version.split(".")[0]
+    # set when current version is found
+    current_found = False
+
+    with open("changelog.md", "r") as file:
+        for line in file:
+            if line.startswith("### "):
+                if not line.startswith(f"### {current_version} ") and not current_found:
+                    # First version in changelog should be the current one
```

Comment:
```suggestion
                    # First version in changelog should be the current one
                    sys.stderr.write(f"Changelog doesn't start with {current_version}\n")
```

or something in those words. Might help debugging :)